pydra/engine/core.py

```diff
@@ -567,31 +583,126 @@ def add(self, task):
             else:
                 task.state = state.State(task.name, other_states=other_states)
         self.node_names.append(task.name)
+        logger.debug(f"Added {task}")
         self.inputs._graph = self.graph_sorted
         return self
 
-    def _run_task(self):
-        for task in self.graph_sorted:
-            # depend on prior tasks that have state
-            task.inputs.retrieve_values(self)
-            if task.state and not hasattr(task.state, "states_ind"):
-                task.state.prepare_states(inputs=task.inputs)
-            if task.state and not hasattr(task.state, "inputs_ind"):
-                task.state.prepare_inputs()
-            if self.plugin is None:
-                task.run()
-            else:
-                from .submitter import Submitter
+    async def run(self, submitter=None, **kwargs):
```

Comment:
i think the base task should be turned async.

Reply:
i.e this function should not be in workflow.

Reply:
@satra this would affect a few things
1) all `Task._run_task()` methods would have to be made async as well
2) `Task`s would no longer be callable
3) Adding to the process pool would not become so straightforward (I don't believe `loop.run_in_executor()` works with coroutines)